File: gaia-sdk-javascript/src/graphql/request/type/UpdatedSkillProvisionImpulse.ts

import {SkillProvision} from "./SkillProvision";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result of a update SkillProvision impulse
 */
export class UpdatedSkillProvisionImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "UpdatedSkillProvisionImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    /**
     * the SkillProvision instance
     */
    public data = (config: (_:SkillProvision) => void) => this.push((registry) => {
        const entity = new SkillProvision();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/UpdatedSkillProvisionImpulse.py
from gaia_sdk.graphql.response.type.SkillProvision import SkillProvision

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class UpdatedSkillProvisionImpulse:
    """
    Impulse which indicates the result of a update SkillProvision impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))

    """
    the SkillProvision instance
    """

    @property
    def data(self) -> SkillProvision:
        return SkillProvision(self.dictionary.get("data"))
